### PR TITLE
DOC: updated wording for ucp_worker_params::thread_mode value

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8543/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@bbenton pls take a look

**Comment by shamisp** [comment]
@bbenton this is port of #3122 to master.

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -746,13 +746,16 @@ typedef struct ucp_worker_params {
...
-     * This value is optional.
-     * If it's not set (along with its corresponding bit in the field_mask -
-     * UCP_WORKER_PARAM_FIELD_THREAD_MODE), the UCS_THREAD_MODE_SINGLE mode
-     * will be used. Note that actual thread mode may be different from mode
-     * passed to @ref ucp_worker_create. To get actual thread mode use
-     * @ref ucp_worker_query.
+     * The parameter thread_mode suggests the thread safety mode which worker
+     * and the associated resources should be created with. This is an
+     * optional parameter. The default value is UCS_THREAD_MODE_SINGLE and
+     * it is used when the value of the parameter is not set. When this
```
Comment (bbenton) [inline src/ucp/api/ucp.h]: I know that this wording was already approved for #3122, but in looking it over again, I have some suggested re-wording.  If this introduces undesirable inconsistencies, then we can stay with the existing wording.

Suggested re-wording plus adding references and emphasis for the parameter name:
The parameter `@a` thread_mode requests the thread safety mode with which the worker and the associated resources should be created. This is an optional parameter. The default value is `@ref` UCS_THREAD_MODE_SINGLE and is used when the value of the parameter is not set.

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -746,13 +746,16 @@ typedef struct ucp_worker_params {
...
+     * parameter along with its corresponding bit in the
+     * field_mask - UCP_WORKER_PARAM_FIELD_THREAD_MODE is set, the
+     * @ref ucp_worker_create attempts to create worker with this thread mode.
+     * The thread mode with which worker is created can differ from the
+     * suggested mode. The actual thread mode of the worker should be obtained
+     * using the query interface @ref ucp_worker_query.
```
Comment (bbenton) [inline src/ucp/api/ucp.h]: Suggested re-wording:
... `@ref` UCP_WORKER_PARAM_FIELD_THREAD_MODE is set, `@ref` ucp_worker_create attempts to create the worker with this thread mode. However, the thread mode with which the worker is created can differ from the requested mode. The actual thread mode of the worker can be obtained...

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (shamisp) [reply src/ucp/api/ucp.h]: @manjugv FYI

File: src/ucp/api/ucp.h (hunk 2 above)
Comment (shamisp) [reply src/ucp/api/ucp.h]: @manjugv FYI

Comment (manjugv) [reply src/ucp/api/ucp.h]: 👍

Comment (hoopoepg) [reply src/ucp/api/ucp.h]: so, @yosefe are you ok with new wording?

**Pull request merged**